NICN1.py

```diff
@@ -0,0 +1,283 @@
+        self.check_alphabet()
+
+        complement_seq = "".join(
+            self._nucleotide_pairs[nucleotide] for nucleotide in self.sequence
+        )
+        return type(self)(complement_seq)
+
+    def get_gc_content(self) -> Union[float, int]:
+        """Return the GC content of the sequence."""
+        self.check_alphabet()
+
+        gc_content = SeqUtils.GC(self.sequence.upper())
+        return gc_content
+
+
+class DNASequence(NucleicAcidSequence, BiologicalSequence, ABC):
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+        self._nucleotide_pairs = {
+            "A": "T",
+            "a": "t",
+            "G": "C",
+            "g": "c",
+            "T": "A",
+            "t": "a",
+            "C": "G",
+            "c": "g",
+        }
+
+    def transcribe(self) -> "RNASequence":
+        """Return the RNA transcript of the sequence."""
+        transcribed_seq = self.sequence.replace("T", "U").replace("t", "u")
+        return RNASequence(transcribed_seq)
+
+
+class RNASequence(NucleicAcidSequence, BiologicalSequence, ABC):
+    """Class representing an RNA sequence."""
+
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+        self._nucleotide_pairs = {
+            "A": "U",
+            "a": "u",
+            "G": "C",
+            "g": "c",
+            "U": "A",
+            "u": "a",
+            "C": "G",
+            "c": "g",
+        }
+
+
+class AminoAcidSequence(BiologicalSequence, ABC):
+    """Class representing an amino acid sequence."""
+
+    def __init__(self, sequence, encoding):
+        self.sequence = sequence.upper()
+        self.encoding = encoding
+        self._residue_names = {
+            "ALA": "A",
+            "ARG": "R",
+            "ASN": "N",
+            "ASP": "D",
+            "CYS": "C",
+            "GLN": "Q",
+            "GLU": "E",
+            "GLY": "G",
+            "HIS": "H",
+            "ILE": "I",
+            "LEU": "L",
+            "LYS": "K",
+            "MET": "M",
+            "PHE": "F",
+            "PRO": "P",
+            "SER": "S",
+            "THR": "T",
+            "TRP": "W",
+            "TYR": "Y",
+            "VAL": "V",
+        }
+        self._residue_mass = {
+            "A": 89,
+            "R": 174,
+            "N": 132,
+            "D": 133,
+            "C": 121,
+            "Q": 146,
+            "E": 147,
+            "G": 75,
+            "H": 155,
+            "I": 131,
+            "L": 131,
+            "K": 146,
+            "M": 149,
+            "F": 165,
+            "P": 115,
+            "S": 105,
+            "T": 119,
+            "W": 204,
+            "Y": 181,
+            "V": 117,
+        }
+
+    def _reformat_based_on_encoding(self) -> Union[str, List[str]]:
+        """Reformat the sequence based on the encoding."""
+        if self.encoding == 1:
+            return self.sequence
+        elif self.encoding == 3:
```

Comment:
Pylint ругается на ошибку с no-else-return (т.к. все условия содержат return). Предложила бы `elif` заменить на `if`:
```suggestion
        if self.encoding == 3:
```